### PR TITLE
Implemented global setups and teardowns, updated docs.

### PR DESCRIPTION
**Pull request by sam-falvo**
Also did a number of refactorings which removed duplication in the code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,11 +53,13 @@ Usage
 
  * **-t, --tests** - Whitespace separated list of test suites to run sequentially
  * **-T, --independent-tests** - Whitespace separated list of test suites to run concurrently
+ * **-m, --max-suites NUMBER** - The number of concurrently executing independent test suites (defaults to 5)
```
Comment (robert-chiniquy) [inline README.md:56]: It's named max but the comment says "the number of", also not sure about 5 as a default.

File: README.md [hunk 2]
```diff
@@ -52,11 +53,13 @@ Usage
...
  * **-ti, --test-init-file** - A path to the initialization file which must export
  `init` function and it is called in a child process *before running the tests in
  each test file
  * **-c, --chdir** - An optional path to which the child process will chdir to before
  running the tests
+ * **-g, --global-setup-teardown STRING** - Specifies the file containing the globalSetUp and globalTearDown procedures.
```
Comment (robert-chiniquy) [inline README.md:62]: This is a path to a js module? Is it required to be fully-qualified? Can it not be?

File: README.md [hunk 3]
```diff
@@ -142,6 +166,20 @@ describe('the bdd module', function(it) {
 });
 ```
 
+A simple example demonstrating how to use global setup and teardown functionality:
+``` javascript
+exports['globalSetUp'] = function(test, assert) {
+  // Set up database schema here...
+  // Push known data set to database here...
```
Comment (robert-chiniquy) [inline README.md:173]: It might be nice to have an actual example here.

File: lib/constants.js [hunk 4]
```diff
@@ -48,8 +48,9 @@ var DEFAULT_VERBOSITY = 2;
 
 var INIT_FUNCTION_NAME = 'init';
 
+var GLOBAL_TEARDOWN_FUNCTION_NAME = 'globalTearDown';
 var TEARDOWN_FUNCTION_NAME = 'tearDown';
-
+var GLOBAL_SETUP_FUNCTION_NAME = 'globalSetUp';
```
Comment (robert-chiniquy) [inline lib/constants.js:53]: For a future direction for Whiskey I'd really like to move away from having magically-named functions and instead use an OO model for tests.

**Comment by robert-chiniquy** [comment]
This looks like a reasonably good way to do what it sets out to do. But why do we want to do that? I worry that this feature reduces isolation between tests suites. It would be great to have a use case for this.

**Comment by sam-falvo** [comment]
Makes sense -- I just wish I could remember!  :-)

**Comment by Kami** [comment]
Awesome pull request. Code looks good to me. 

I will work on merging it and doing a final review before publishing a new version to npm in a bit.

**Comment by Kami** [comment]
I just want to add that in general I do agree with @robert-chiniquy and in the future we should be more careful when accepting and adding new features.

One of the main "problems" with Whiskey is that is has too many features and a complex UI (command line interface) with too many options which makes it confusing and not so easy to use.

One of my long term goals is to clean everything up, remove unused, add better documentation and make it more user friendly.

On a flip side, Whiskey is one of those libraries which usually just works so I don't have so much motivation to start tearing it up... :)

**Comment by Kami** [comment]
I've published a new version (0.7.0) to npm - https://npmjs.org/package/whiskey/0.7.0

**Pull request merged**